civis/futures.py

```diff
@@ -495,35 +484,27 @@ class _ContainerShellExecutor(_CivisExecutor):
     inc_script_names: bool, optional
         If ``True``, a counter will be added to the ``script_name`` to create
         the script names for each submission.
+    **kwargs:
+        Additional keyword arguments will be passed
+        directly to ``scripts.post_containers``.
 
     See Also
     --------
     civis.APIClient.scripts.post_containers
     """
     def __init__(self, docker_image_name="civisanalytics/datascience-python",
                  docker_image_tag="latest",
-                 repo_http_uri=None,
-                 repo_ref=None,
-                 git_credential_id=None,
                  script_name=None,
                  required_resources=None,
```

Comment:
The `required_resources` and `docker_image_name` are required arguments to `post_containers`, not optional keyword parameters, so it doesn't seem inconsistent to treat those differently. The bigger reason I left them was that I wanted to control the defaults and keep the defaults explicit. `hidden` defaults to `True` here, which is different from the Platform default. The default on `docker_image_tag` happens to match the Platform default, but that's important enough I'd rather have it explicit.

`script_name` maybe could be left to keyword arguments. Looking at it, I at least need to improve the docstring. The parameter name in `post_containers` is "name", and it's a bit ambiguous to pass a "name" parameter here. Does it mean the name of the executor? I could go either way on that one.